File: Integraupt-Backend/backend-horarios-laravel/app/Enums/DiaSemana.php
<?php

namespace App\Enums;

/**
 * Replica del enum Java: com.horarios.model.DiaSemana
 */
enum DiaSemana: string
{
    case LUNES = 'Lunes';
    case MARTES = 'Martes';
    case MIERCOLES = 'Miercoles';
    case JUEVES = 'Jueves';
    case VIERNES = 'Viernes';
    case SABADO = 'Sabado';

    /**
     * Busca un DiaSemana por nombre (case-insensitive).
     * Replica: DiaSemana.fromNombre(String nombre)
     */
    public static function fromNombre(string $nombre): self
    {
        foreach (self::cases() as $dia) {
            if (strcasecmp($dia->value, $nombre) === 0) {
                return $dia;
            }
        }
        throw new \InvalidArgumentException("Día de la semana no válido: {$nombre}");
    }

    /**
     * Retorna todos los días en orden.
     */
    public static function ordered(): array
    {
        return [
            self::LUNES,
            self::MARTES,
            self::MIERCOLES,
            self::JUEVES,
            self::VIERNES,
            self::SABADO,
        ];
    }
}

File: integraupt-web/Integraupt-Backend/BackendHorarios/src/main/java/com/horarios/model/DiaSemana.java
package com.horarios.model;

import java.util.Arrays;

public enum DiaSemana {
    LUNES("Lunes"),
    MARTES("Martes"),
    MIERCOLES("Miercoles"),
    JUEVES("Jueves"),
    VIERNES("Viernes"),
    SABADO("Sabado");

    private final String nombre;

    DiaSemana(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return nombre;
    }

    public static DiaSemana fromNombre(String nombre) {
        return Arrays.stream(values())
                .filter(dia -> dia.nombre.equalsIgnoreCase(nombre))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("Día de la semana no válido: " + nombre));
    }
}